mlflow/tracing/provider.py

```diff
@@ -114,6 +114,34 @@ def _force_set_otel_tracer_provider(tracer_provider):
     trace.set_tracer_provider(tracer_provider)
 
 
+def _catch_exception(msg: str, default_return=None):
```

Comment:
```suggestion
def _catch_exception(msg: str, default_return: Optional[Any]=None):
```

because `msg` has a type hint?

Reply:
Talked offline, will address in the follow-up refactoring w.r.t Serena's comment, to unblock release first🙂